mock_controller_driver/include/mock_controller_driver/mock_controller_driver_node.hpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#pragma once
+
+#include <rclcpp/rclcpp.hpp>
+#include <functional>
+#include <std_msgs/msg/string.hpp>
+#include <std_srvs/srv/empty.hpp>
+
+#include <carma_driver_msgs/srv/set_enable_robotic.hpp>
+#include <carma_driver_msgs/msg/robot_enabled.hpp>
+#include <autoware_msgs/msg/vehicle_cmd.hpp>
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+
+namespace mock_controller_driver
+{
+
+  /**
+   * \brief TODO for USER: Add class description
+   *
+   */
+  class MockControllerDriver : public carma_ros2_utils::CarmaLifecycleNode
+  {
+
+  private:
+    // Subscribers
+    carma_ros2_utils::SubPtr<autoware_msgs::msg::VehicleCmd> vehicle_cmd_sub_;
+
+    // Publishers
+    carma_ros2_utils::PubPtr<carma_driver_msgs::msg::RobotEnabled> robot_status_pub_;
+
+
+    // Service Servers
+    carma_ros2_utils::ServicePtr<carma_driver_msgs::srv::SetEnableRobotic> enable_robotic_srvice_;
+
+    // Timers
+    rclcpp::TimerBase::SharedPtr spin_timer_;
+
+
+  public:
+
+      // Pub
+    const std::string robot_status_topic_ = "controller/robot_status";
+
+    // Sub
+    const std::string vehicle_cmd_topic_ = "vehicle_cmd";
+
+    // Service
+    const std::string enable_robotic_srv_ = "controller/enable_robotic";
+
+    // Robot Status flags
+    bool robot_active_ = false;
+    bool robot_enabled_ = false;
```

Comment:
What is the difference between the two?
for example, which one is for when the vehicle is engaged? I am struggling to understand the [comment on the msg itself too](https://github.com/usdot-fhwa-stol/carma-msgs/blob/develop/carma_driver_msgs/msg/RobotEnabled.msg#L7C1-L11C19)

Could we give more descriptive or commonly used term by the team?

Reply:
From what I understood from this driver and guidance plugin, the setRobotEnable service call is when the vehicle is first engaged, and robot_active_ is when vehicle is engaged and receiving commands.
I couldn't find any documentation to explain the sequence. Even the guidance plugin description are brief and kind of vague.

Reply:
added a sentence to the robot enable topic.

Reply:
yeah that's my understanding as well. I am just getting confused by the order of the service call and vehicle_cmd subscription order. I would think vehicle_cmd topic subscription will only receive values (unless it normally receives speed=0 commands constantly) after the vehicle is engaged (the enable service call). But I could be wrong, so I am approving it and we can integration test it later.